numpy/core/src/umath/ufunc_object.c

```diff
@@ -565,7 +565,9 @@ get_ufunc_arguments(PyUFuncObject *ufunc,
                     PyObject **out_typetup,
                     int *out_subok,
                     PyArrayObject **out_wheremask,
-                    PyObject **out_axes)
+                    PyObject **out_axes,
+                    PyObject **out_axis,
+                    int *out_keepdims)
```

Comment:
Any reason not to use a `npy_bool` for this and `out_subok`?

Reply:
In the main routine, it is `int`, though, and I use `-1` as a value to indicate "unitialized". Overall, perhaps best done in separate cleanup.